repair.py

```diff
@@ -0,0 +1,162 @@
+import asyncio
+import logging
+import sys
+import tempfile
+
+import createrepo_c as cr
+import shutil
+from django.core.files.storage import default_storage as storage
+
+from pulpcore.plugin.models import (
+    ContentArtifact,
+    RepositoryVersion,
+)
+
+from pulp_rpm.app.models import (
+    Package,
+    RpmPublication,
+    RpmRemote,
+    RpmRepository,
+    RpmDistribution,
+)
+
+log = logging.getLogger(__name__)
+loop = asyncio.get_event_loop()
+
+broken_criteria = {
+    "files": [],
+    "changelogs": [],
+}
+broken_packages = Package.objects.filter(**broken_criteria)
+
+print("""
+##########################################
+#  Repair tool for Pulp RPM issue #9107  #
+#    https://pulp.plan.io/issues/9107    #
+##########################################
+""")
+
+print(
+"""Analysis
+========
+"""
+)
+
+total_packages = Package.objects.all().count()
+total_broken_packages = broken_packages.count()
+print("Total broken packages: {}/{}".format(total_broken_packages, total_packages))
+
+broken_content_artifacts = ContentArtifact.objects.filter(content__in=broken_packages)
+assert broken_content_artifacts.count() == total_broken_packages
+
+broken_content_artifacts_with_files = broken_content_artifacts.filter(artifact__isnull=False)
+broken_content_artifacts_without_files = broken_content_artifacts.filter(artifact__isnull=True)
+
+total_broken_packages_with_artifacts = broken_content_artifacts_with_files.count()
+print("Broken packages with local RPMs available: {}/{}".format(total_broken_packages_with_artifacts, total_broken_packages))
```

Comment:
How about `broken package metadata with ...`? Just to be clear which part is off, `a broken package` sounds more like a corrupted RPM to me...but maybe it's just me :)

Reply:
Good suggestion